pymatgen/entries/computed_entries.py

```diff
@@ -494,7 +500,7 @@ def __str__(self):
         return self.__repr__()
 
     @classmethod
-    def from_dict(cls, d) -> "ComputedEntry":
+    def from_dict(cls, d: dict) -> "ComputedEntry":
```

Comment:
Found a small bug in `ComputedEntry.from_dict`, not related to your changes. Since we're making a lot of changes to `ComputedEntry` here, would you mind changing line 511 to

```
if d.get("correction", 0) != 0 and not d.get("energy_adjustments"):
```

Otherwise de-serializing `ComputedEntry` that don't have a .correction` attribute fails.